datafusion/expr/src/expr.rs

```diff
@@ -869,53 +869,56 @@ impl Expr {
     }
 }
 
-/// Format expressions for display as part of a logical plan. In many cases, this will produce
-/// similar output to `Expr.name()` except that column names will be prefixed with '#'.
-impl fmt::Display for Expr {
-    fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
-        write!(f, "{self:?}")
-    }
+#[macro_export]
```

Comment:
Given how common this functionality is I wonder if we could move this macro (or maybe make a function as I don't see any benefit in making it a macro)  in `datafusion_util` and use it more around the codebase.

Just a thought

Reply:
I think making a function and moving to `datafusin_util` is a good option.

Reply:
I can give it a try maybe after this PR is merged